### PR TITLE
feat: CFD-244 Save group passenger configs

### PR DESCRIPTION
**Pull request by jacsamell**
# Description

-   Save group passenger config in the DB

# Testing instructions

-   Create a group passenger type and name it
-   Check it saves in the DB
-   Create a group without a name and check it doesn't save

# Type of change

-   [x] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

# Checklist:

-   [x] Able to run pr locally
-   [x] Followed acceptance criteria
-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have made corresponding changes to the documentation if applicable
-   [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: src/pages/api/definePassengerType.ts [hunk 1]
```diff
@@ -301,6 +302,11 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
                             }`,
                         );
                     } else {
+                        const trimmedName = passengerTypeName?.trim();
+                        if (trimmedName) {
+                            await insertPassengerType(getAndValidateNoc(req, res), companions, trimmedName, true);
```
Comment (RobCatton) [inline src/pages/api/definePassengerType.ts:307]: Do we need to do a check here to see if there is already a passenger type with the same name? And if so, pass an error message back and redirect back to the page.

File: tests/pages/api/definePassengerType.test.ts [hunk 2]
```diff
@@ -493,6 +494,56 @@ describe('definePassengerType', () => {
         });
         await definePassengerType(req, res);
         expect(upsertPassengerTypeSpy).not.toBeCalled();
+        expect(insertPassengerTypeSpy).not.toBeCalled();
         expect(updateSessionAttributeSpy).toBeCalledWith(req, GROUP_PASSENGER_INFO_ATTRIBUTE, expect.anything());
     });
+
+    it('should save the passenger types for groups if a name is provided', async () => {
```
Comment (RobCatton) [inline tests/pages/api/definePassengerType.test.ts:501]: If we do the above (adding check to see if duplicate name) we might need a unit test here to check that, also.

File: src/pages/api/definePassengerType.ts (hunk 1 above)
Comment (jacsamell) [reply src/pages/api/definePassengerType.ts:307]: Good shout

File: tests/pages/api/definePassengerType.test.ts (hunk 2 above)
Comment (jacsamell) [reply tests/pages/api/definePassengerType.test.ts:501]: Done 👍

**Review by RobCatton — APPROVED** [review]

**Pull request merged**